### ISSUE TITLE
[Android] Allow background audio along with PPSSPP

### ISSUE THREAD

**Issue by fefo-dev** [hrydgard/ppsspp#13078]
To my knowledge, PPSSPP will take audio focus of any sound source, like Youtube, unless either PPSSPP is muted or the source has any way around it (a few have). Is it possible to add something like a toogle to make so both can be played together, or it's a limitation within the emulator's sound engine? I wonder since if that was the case it should pause other sources even when muted.

Much appreciated.

**Comment by hrydgard** [comment]
The emulator's sound engine has no particular limitations (except that it's still using OpenSL, the older (but not the oldest) Android audio API, instead of AAudio). 

If you want this to happen faster, please go research what specific API calling changes I need to make, because that research is probably the main time sink here.

**Comment by fefo-dev** [comment]
I was mistaken,  at least now I can't retain audio focus in game despite testing with either OpenSL, AAudio or AudioTrack as per AIMP (music player) output options.

Sadly I'm unskilled to help outside of testing other apps: Snes9x EX+ (https://github.com/Rakashazi/emu-ex-plus-alpha) takes focus on the first load, but never again after I restart playing a song, and M64Plus FZ (https://github.com/fzurita/mupen64plus-ae/tree/fz-master) does not take focus at any moment.

Thanks for the quick answer and explanation.

**Comment by strikersix23** [comment]
I believe it might be a flag as audio focus has been around for a long while ill see what permissions are required

**Comment by strikersix23** [comment]
Yeah its been around since android 2.3

**Comment by strikersix23** [comment]
Reference
https://developer.android.com/guide/topics/media-apps/audio-focus

**Comment by unknownbrackets** [comment]
We are calling an older (older than Android 8) API to gain audio focus, which indicates that we don't support "ducking" (reducing volume while other apps play notifications or voice.)  Maybe that would help here.

-[Unknown]

**Comment by Unknowncat1991** [comment]
<img width="2110" height="314" alt="Image" src="https://github.com/user-attachments/assets/64a71187-cc18-4e54-9656-48561607f7d0" />
This was already implemented?

**Comment by fefo-dev** [comment]
Is that on latest stable? I'm on 1.20.4 and there's no option:
[https://i.imgur.com/mVmDlxV.png](url)

**Comment by hrydgard** [comment]
No, latest development build, so it will be in 1.21, and if you can't wait, https://www.ppsspp.org/devbuilds/ . You could also join the beta program, I will release a new version to the beta channel in a couple of weeks.

https://www.ppsspp.org/docs/development/beta-testing/
